app/services/courses_query.rb

```diff
@@ -22,6 +23,37 @@ def initialize(params:)
   end
 
   def call
-    @scope.distinct
+    @scope = visa_sponsorship_filter
+    @scope = @scope.distinct
+
+    log_query_info
+
+    @scope
+  end
+
+  def visa_sponsorship_filter
+    return @scope if params[:can_sponsor_visa].blank?
+
+    filter_applied(name: :can_sponsor_visa)
```

Comment:
`add_filter`?

Reply:
I removed this as per suggestion https://github.com/DFE-Digital/publish-teacher-training/pull/4751/files#r1875757673